### DISCUSSION TITLE
Migration to v18 -> @ngrx/signals: SignalStore members cannot be overridden. Trying to override... how to solve?

### DISCUSSION THREAD

**Discussion by TomaszStanik** [ngrx/platform#4520] [Q&A]
Hi!

I have 3 signalStoreFeatures.

A, B, C

A got methods `refresh` (and many other logic)
```ts
withA() {
   return signalStoreFeature(
      withMethods(() => {
         return {
            refresh(): void { 
               // ...
            },
      }
}
```
B inherits A and overrides `refresh` method (needs to pass extra params to method)
```ts
withB() {
   return signalStoreFeature(
     withA(),
      withMethods(store => {
         return {
            refresh(myParams: RefreshParams): void { 
               // do a extra logic myParams
               // refresh method from A should not be visible to call
            },
      }
}
```
C inherits B and overrides `refresh` method (needs to pass extra params to method)
```ts
withC() {
   return signalStoreFeature(
     withB(),
      withMethods(store => {
         return {
            refresh(myParams: RefreshParams, config: MyConfig): void { 
                // 
            },
      }
}
```

How to deal with that? This is how my few features worked and now I'm noticing many warnings.

I found this topic: https://github.com/ngrx/platform/pull/4199

but `  {allowOverrides: true},` I could not use in signalStoreFeature.
![image](https://github.com/user-attachments/assets/d0c1c6e5-b7b5-4c40-81cf-948cbecddf01)

**Comment by rainerhahnekamp** [comment]
@TomaszStanik, so the SignalStore's features are not meant to be overriden. We don't see these features as class inheritance. You should apply "composition over inheritance" instead.

Can you design your SignalStore so that the features don't expose `refresh` but use a different name? Your final store could then provide the actual `refresh` method that calls the `refresh*` from the features.

Do you understand what I mean?
